src/anomalib/models/dsr/torch_model.py

```diff
@@ -0,0 +1,1196 @@
+"""PyTorch model for the DSR model implementation."""
+
+# Original Code
+# Copyright (c) 2022 VitjanZ
+# https://github.com/VitjanZ/DSR_anomaly_detection.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Modified
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import Callable, Dict
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor, nn
+
+
+class DsrModel(nn.Module):
+    """DSR PyTorch model consisting of the discrete latent model, image reconstruction network,
+    subspace restriction modules, anomaly detection module and upsampling module.
+
+    Args:
+        embedding_dim (int): Dimension of codebook embeddings.
+        num_embeddings (int): Number of embeddings.
+        anom_par (float):
```

Comment:
Is this intentionally left blank?

Reply:
Also, as raised above, `anom_par` is not so clear to understand. It would be great if it is renamed

Reply:
See above